core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -227,6 +226,8 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
                 thisAggDocCountError = 0;
             } else if (InternalOrder.isCountDesc(this.order)) {
                 thisAggDocCountError = terms.getBucketsInternal().get(terms.getBucketsInternal().size() - 1).docCount;
```

Comment:
This line does not look correct to me: what if the `terms` aggregation already has an error? And what if the {size-1}-th terms has some error too?